opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/CalciteScriptEngine.java

```diff
@@ -206,7 +207,7 @@ public org.apache.calcite.linq4j.tree.Expression field(
      */
     private Expression tryConvertDocValue(Expression docValueExpr, ExprType exprType) {
       return switch (exprType) {
-        case INTEGER, SHORT -> EnumUtils.convert(docValueExpr, Long.class);
+        case INTEGER, SHORT, BYTE -> EnumUtils.convert(docValueExpr, Long.class);
```

Comment:
this is the code change to fix the bug. The rests are protected programming which is not related to this bug.